packages/react/src/components/dropdown-list/dropdown-list.tsx

```diff
@@ -204,426 +140,258 @@ export const DropdownList: VoidFunctionComponent<DropdownListProps> = ({
-        () => {
-            if (!shouldFilterOptions) {
-                return options;
-            }
-            return filterOptions(options, searchValue);
-        },
-        [options, searchValue, shouldFilterOptions],
-    );
 
-    const findOptionByValue: (needle?: string) => Option | undefined = useCallback(
-        (needle) => options.find((option) => option.value === needle),
-        [options],
-    );
-    const [inputValue, setInputValue] = useState(() => findOptionByValue(selectedOptionValue)?.label ?? '');
-
-    useEffect(() => {
-        const wantedOption = findOptionByValue(value);
-        if (wantedOption) {
-            setSelectedOptionValue(wantedOption.value);
-            setInputValue(wantedOption.label);
-        } else if (skipOption && skipOption.value === value) {
-            setSelectedOptionValue(skipOption.value);
-            setInputValue('');
-        }
-    }, [findOptionByValue, options, skipOption, skipSelected, value]);
-
-    const handleOpen: () => void = useCallback(() => {
-        if (!disabled) {
-            if (!open) {
-                setFocus(true);
-                if (searchable && inputRef.current) {
-                    inputRef.current.focus();
-                    setFocusedValue('');
-                } else {
-                    setFocusedValue(selectedOptionValue);
-                }
-            } else {
-                inputRef.current?.focus();
-                setAutofocus(false);
-                setFocusedValue('');
-            }
-            setOpen(!open);
-        }
-    }, [disabled, open, searchable, selectedOptionValue]);
-
-    const handleClickOutside: (event: MouseEvent) => void = useCallback((event) => {
-        const clickIsOutside = !eventIsInside(event, wrapperRef.current, listboxRef.current);
-        const shouldClose = (wrapperRef.current === null || clickIsOutside) && open;
-
-        if (shouldClose) {
-            handleOpen();
-            inputRef.current?.blur();
-        }
-    }, [open, handleOpen]);
-
-    useEffect(() => {
-        document.addEventListener('mouseup', handleClickOutside);
-        return () => document.removeEventListener('mouseup', handleClickOutside);
-    }, [handleClickOutside, open]);
+    const [open, setOpen] = useState(defaultOpen);
 
-    function findOption(optionsArray: Option[], findValue: string): Option | undefined {
-        return optionsArray.find((option) => option.label.toLowerCase() === findValue.toLowerCase());
+    function findOptionByValue(searchValue?: string): DropdownListOption | undefined {
+        return options.find((option) => option.value === searchValue);
     }
 
-    function focusFirstElementFromArray(array: Option[]): void {
-        const firstArrayElement = array.find((element) => !element.disabled);
+    const [selectedOption, setSelectedOption] = useState<DropdownListOption | undefined>(
+        () => findOptionByValue(value || defaultValue) ?? options[0],
```

Comment:
Il y a un petit bug quand la première option est disabled, comme dans la denière story du storybook. Je pense que ça règlerait le problème.
```suggestion
        () => findOptionByValue(value || defaultValue) ?? options.find((element) => !element.disabled),
```
Tu pourrais même réutiliser la fonction de predicate si tu l'extract.

Reply:
Oh, bien vu! J'ai corrigé pour qu'une option _disabled_  ne soit pas sélectionnée comme valeur par défaut. Par contre, j'ai laissé la possibilité de sélectionner n'importe quelle valeur avec `defaultValue` ou `value` (voir l'autre réponse plus bas).